### PR TITLE
Add workaround for Meltdown bug in CaaSP

### PR DESCRIPTION
**Pull request by drpaneas**
For more information read bsc#1074836 and poo#30026

- Related ticket: https://progress.opensuse.org/issues/30026
- Needles: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/634
- Verification run: http://skyrim.qam.suse.de/tests/overview?distri=caasp&version=2.0&build=%3A6053%3Avelum.meltdown6&groupid=98

### CODE REVIEW

File: tests/caasp/stack_bootstrap.pm [hunk 1]
```diff
@@ -130,8 +130,17 @@ sub kubectl_config {
 }
 
 sub run {
-    assert_screen 'velum-bootstrap-page', 90;
     barrier_wait {name => "WORKERS_INSTALLED", check_dead_job => 1};
+    if (check_screen('velum-sorry', 300)) {
```
Comment (kravciak) [inline tests/caasp/stack_bootstrap.pm]: check_screen has one negative side-effect - if it does not match, you have to wait the timeout (300s).
You expect bug#1074836 and if it's not there you delay for 5 minutes (all QA tests). You can wait for multiple needles this way:
```
assert_screen [qw(velum-bootstrap-page velum-sorry)], 90;
if (match_has_tag 'velum-sorry') {
 ...workaround...
}
```

File: tests/caasp/stack_bootstrap.pm [hunk 2]
```diff
@@ -130,8 +130,17 @@ sub kubectl_config {
 }
 
 sub run {
-    assert_screen 'velum-bootstrap-page', 90;
     barrier_wait {name => "WORKERS_INSTALLED", check_dead_job => 1};
```
Comment (kravciak) [inline tests/caasp/stack_bootstrap.pm]: Please don't switch order here, otherwise we won't know how long it takes to load bootstrap page.

Comment (drpaneas) [reply tests/caasp/stack_bootstrap.pm]: ok, I did that on purpose actually in order to benefit from the time during the `barrier_wait`, but I see your point ;) I will re-order it.

**Comment by kravciak** [comment]
Optionally you can create sub workaround_bsc1074836 in lib/caasp_controller.pm to not repeat the code - but again, that's up to you.

File: tests/caasp/stack_bootstrap.pm (hunk 1 above)
Comment (drpaneas) [reply tests/caasp/stack_bootstrap.pm]: Thanks for the tip. I didn't know that.

File: tests/caasp/stack_bootstrap.pm [hunk 3]
```diff
@@ -130,8 +130,17 @@ sub kubectl_config {
...
+    if (check_screen('velum-sorry', 300)) {
+        record_soft_failure('bnc#1074836 - delay caused due to Meltdown');
+        # workaround for meltdown
+        for (my $i=0; $i <= 9; $i++) {
```
[the remark this answers is not in the document]
Comment (drpaneas) [reply tests/caasp/stack_bootstrap.pm]: Less lines, I like it.

File: tests/caasp/stack_update.pm [hunk 4]
```diff
@@ -64,7 +64,13 @@ sub run {
     assert_and_click 'velum-update-reboot';
 
     # Update all nodes - this part takes long time (~2 minutes per node)
-    assert_screen "velum-$nodes-nodes-outdated", 300;
+    assert_screen [qw(velum-$nodes-nodes-outdated velum-sorry)], 300;
+    if (match_has_tag 'velum-sorry') {
+        record_soft_failure('bnc#1074836 - delay caused due to Meltdown');
+        # workaround for meltdown
+        send_key_until_needlematch 'velum-$nodes-nodes-outdated', 'f5', 10, 120;
```
Comment (kravciak) [inline tests/caasp/stack_update.pm]: I believe you need double quotes around 'velum-$nodes-nodes-outdated', not singles

Comment (drpaneas) [reply tests/caasp/stack_update.pm]: True. I am restarting the test locally to see if it passes.

**Comment by okurz** [comment]
LGTM

**Comment by drpaneas** [comment]
Local verification just passed: http://skyrim.qam.suse.de/tests/1359# :D

**Pull request merged**